src/main/java/com/autotune/analyzer/services/CreateExperiment.java

```diff
@@ -81,21 +82,26 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
         try {
             String inputData = request.getReader().lines().collect(Collectors.joining());
             List<CreateExperimentSO> experimentSOList = Arrays.asList(new Gson().fromJson(inputData, CreateExperimentSO[].class));
-            List<KruizeObject> kruizeExpList = new ArrayList<>();
-            for (CreateExperimentSO createExperimentSO: experimentSOList) {
-                KruizeObject kruizeObject = Utils.Converters.KruizeObjectConverters.convertCreateExperimentSOToKruizeObject(createExperimentSO);
-                if (null != kruizeObject) {
-                    kruizeExpList.add(kruizeObject);
-                }
-            }
-            new ExperimentInitiator().validateAndAddNewExperiments(mainKruizeExperimentMap, kruizeExpList);
-            //TODO: UX needs to be modified - Handle response for the multiple objects
-            KruizeObject invalidKruizeObject = kruizeExpList.stream().filter((ko) -> (!ko.getValidationData().isSuccess())).findAny().orElse(null);
-            if (null == invalidKruizeObject) {
-                sendSuccessResponse(response, "Experiment registered successfully with Kruize.");
+            if (experimentSOList.size() > 1) {
+                LOGGER.error(AnalyzerErrorConstants.AutotuneObjectErrors.UNSUPPORTED_EXPERIMENT);
+                sendErrorResponse(response, null, HttpServletResponse.SC_BAD_REQUEST, AnalyzerErrorConstants.AutotuneObjectErrors.UNSUPPORTED_EXPERIMENT );
             } else {
-                LOGGER.error("Failed to create experiment due to {}", invalidKruizeObject.getValidationData().getMessage());
-                sendErrorResponse(response, null, HttpServletResponse.SC_BAD_REQUEST, invalidKruizeObject.getValidationData().getMessage());
+                List<KruizeObject> kruizeExpList = new ArrayList<>();
+                for (CreateExperimentSO createExperimentSO : experimentSOList) {
+                    KruizeObject kruizeObject = Utils.Converters.KruizeObjectConverters.convertCreateExperimentSOToKruizeObject(createExperimentSO);
+                    if (null != kruizeObject) {
+                        kruizeExpList.add(kruizeObject);
+                    }
+                }
+                new ExperimentInitiator().validateAndAddNewExperiments(mainKruizeExperimentMap, kruizeExpList);
+                //TODO: UX needs to be modified - Handle response for the multiple objects
+                KruizeObject invalidKruizeObject = kruizeExpList.stream().filter((ko) -> (!ko.getValidationData().isSuccess())).findAny().orElse(null);
+                if (null == invalidKruizeObject) {
+                    sendSuccessResponse(response, "Experiment registered successfully with Kruize.");
+                } else {
+                    LOGGER.error("Failed to create experiment due to {}", invalidKruizeObject.getValidationData().getMessage());
```

Comment:
Message should be
`Failed to create experiment: {}"`

Reply:
Done